src/module.c

```diff
@@ -7146,6 +7154,24 @@ int moduleAllModulesHandleReplAsyncLoad(void) {
     return 1;
 }
 
+int moduleVerifyAllModulesAllowAtomicSlotMigrationOrReply(client *c) {
+    dictIterator *di = dictGetIterator(modules);
+    dictEntry *de;
+
+    while ((de = dictNext(di)) != NULL) {
+        struct ValkeyModule *module = dictGetVal(de);
+        if (!(module->options & VALKEYMODULE_OPTIONS_ALLOW_ATOMIC_SLOT_MIGRATION)) {
+            addReplyErrorFormat(c, "The module %s does not support atomic slot migrations. "
+                                   "Please ensure all modules have declared support for "
+                                   "atomic slot migration and try again",
+                                module->name);
+            return C_ERR;
+        }
+    }
+    dictReleaseIterator(di);
+    return C_OK;
+}
```

Comment:
* Shall we just verify here if all the modules support ASM or not and add the error reply on the caller end ? With the current output format we don't guarantee we are able to inform the client about all the modules which  don't support ASM.
* And I think you're just following the code structure but do we need to prepend the method name with `module` ?